packages/grid/x-data-grid-pro/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  GridStateColDef,
+  GridRowId,
+  GridColDef,
+  GridValueOptionsParams,
+  GridValueFormatterParams,
+  GridApi,
+} from '@mui/x-data-grid';
+import { buildWarning } from '@mui/x-data-grid/internals';
+import { GridExceljsProcessInput } from '../gridExcelExportInterface';
+
+const getExcelJs = () => import('exceljs');
+
+const warnInvalidFormattedValue = buildWarning([
+  'MUI: When the value of a field is an object or a `renderCell` is provided, the Excel export might not display the value correctly.',
+  'You can provide a `valueFormatter` with a string representation to be used.',
+]);
+
+const getFormattedValueOptions = (
+  colDef: GridColDef,
+  { id, row, field }: GridValueOptionsParams,
+  api: GridApi,
+) => {
+  if (!colDef.valueOptions) {
+    return [];
+  }
+  let valueOptionsFormatted;
+  if (typeof colDef.valueOptions === 'function') {
+    valueOptionsFormatted = colDef.valueOptions({ id, row, field });
+  } else {
+    valueOptionsFormatted = colDef.valueOptions;
+  }
+
+  if (colDef.valueFormatter) {
+    valueOptionsFormatted = valueOptionsFormatted.map((option) => {
+      if (typeof option === 'object') {
+        return option;
+      }
+
+      const params: GridValueFormatterParams = { field, api, value: option };
+      return String(colDef.valueFormatter!(params));
+    });
+  }
+  return valueOptionsFormatted.map((option) =>
+    typeof option === 'object' ? option.label : option,
+  );
+};
+
+const serializeRow = (id: GridRowId, columns: GridStateColDef[], api: GridApi) => {
+  const row = {};
+  const dataValidation = {};
+
+  const firstCellParams = api.getCellParams(id, columns[0].field);
+  const outlineLevel = firstCellParams.rowNode.depth;
+
+  columns.forEach((column) => {
+    const cellParams = api.getCellParams(id, column.field);
+    switch (cellParams.colDef.type) {
+      case 'singleSelect': {
+        const valueOptions: GridValueOptionsParams = {
+          id: cellParams.id,
+          row: cellParams.row,
+          field: cellParams.field,
+        };
+        const formattedValueOptions = getFormattedValueOptions(
+          cellParams.colDef,
+          valueOptions,
+          api,
+        );
+        dataValidation[column.field] = {
+          type: 'list',
+          allowBlank: true,
+          formulae: [`"${formattedValueOptions.map((x) => x.toString()).join(',')}"`],
+        };
+        const formattedValue = api.getCellParams(id, column.field).formattedValue;
+        if (process.env.NODE_ENV !== 'production') {
+          if (String(cellParams.formattedValue) === '[object Object]') {
+            warnInvalidFormattedValue();
+          }
+        }
+        row[column.field] =
+          typeof formattedValue === 'object' ? formattedValue.label : formattedValue;
+        break;
+      }
+      case 'boolean':
+      case 'number':
+      case 'date':
+      case 'dateTime':
+        row[column.field] = api.getCellParams(id, column.field).value;
```

Comment:
The `boolean` column, for example, has "yes" and "no" as labels but it exports using the [`TRUE`](https://support.microsoft.com/en-us/office/true-function-7652c6e3-8987-48d0-97cd-ef223246b3fb) function from Excel. I don't think users are used to use this function to denote boolean values.

```suggestion
        row[column.field] = api.getCellParams(id, column.field).formattedValue;
```

Reply:
That's the way ExcelJS handle boolean
https://github.com/exceljs/exceljs#boolean-value

I tried `row[column.field] = "TRUE"` and `row[column.field] = "=True"` but it is always considered as a string and not the boolean value

Reply:
We can wait for feedback from the community. The way you did is also correct. I only think that users will always convert from `TRUE` to another value that makes more sense. If I open the Excel export in Google Spreadshets or Excel it displays as follow:

![image](https://user-images.githubusercontent.com/42154031/157042675-f42795d1-0142-40fe-b7ab-a78ff0f07cb1.png)

If I create this same spreadsheet by hand I would've used "Y"/"N" or "Yes"/"No" instead. Anyway, I think someone will ask for an option to convert boolean values to a custom string value.